devito/ir/support/utils.py

```diff
@@ -81,7 +81,12 @@ def build_iterators(mapper):
         for d, offs in v.items():
             if d.is_Stepping:
                 values = iterators.setdefault(d.parent, [])
-                for i in sorted(offs):
+                # Offsets are sorted so that the semantic order (t0, t1, t2)
+                # follows sympy's index ordering (t, t-1, t+1) afer modulo replacement
+                # and guarantying bitwise reproducibility.
```

Comment:
I would drop this, or at least relax it -- isn't it too strong?

Reply:
Isn't this just revisiting the discussion with @mloubout above?

Reply:
I think he just means drop "guarantying bitwise reproducibility." that is a bit too strong of a statement.

Reply:
Changed last line to 'so that associativity errors are consistent.'